ipinfo/core.go

```diff
@@ -0,0 +1,254 @@
+package ipinfo
+
+import (
+	"net"
+)
+
+// Core represents data from the Core API.
+type Core struct {
+	IP       net.IP       `json:"ip"`
+	Hostname string       `json:"hostname"`
+	City     string       `json:"city"`
+	Region   string       `json:"region"`
+	Country  string       `json:"country"`
+	Location string       `json:"loc"`
+	Org      string       `json:"org"`
+	Postal   string       `json:"postal"`
+	Timezone string       `json:"timezone"`
+	ASN      *CoreASN     `json:"asn"`
+	Company  *CoreCompany `json:"company"`
+	Carrier  *CoreCarrier `json:"carrier"`
+	Privacy  *CorePrivacy `json:"privacy"`
+	Abuse    *CoreAbuse   `json:"abuse"`
+	Domains  *CoreDomains `json:"domains"`
+}
+
+// CoreASN represents ASN data for the Core API.
+type CoreASN struct {
+	ASN    string `json:"asn"`
+	Name   string `json:"name"`
+	Domain string `json:"domain"`
+	Route  string `json:"route"`
+	Type   string `json:"type"`
+}
+
+// CoreCompany represents company data for the Core API.
+type CoreCompany struct {
+	Name   string `json:"name"`
+	Domain string `json:"domain"`
+	Type   string `json:"type"`
+}
+
+// CoreCarrier represents carrier data for the Core API.
+type CoreCarrier struct {
+	Name string `json:"name"`
+	Mcc  string `json:"mcc"`
```

Comment:
Should be `MCC` and `MNC` (all upper case), or even better `MobileCountryCode` and `MobileNetworkCode`